python/ray/dag/compiled_dag_node.py

```diff
@@ -649,12 +649,12 @@ def _get_or_compile(
 
             if isinstance(task.dag_node, ClassMethodNode):
                 readers = [self.idx_to_task[idx] for idx in task.downstream_node_idxs]
-                assert len(readers) == 1
```

Comment:
This assertion will make the following case fail: the actor a has two readers (actor b and itself)

```python
a = Actor.remote(0)
b = Actor.remote(100)
with InputNode() as inp:
    dag = a.inc.bind(inp)
    dag = MultiOutputNode([a.inc.bind(dag), b.inc.bind(dag)])

compiled_dag = dag.experimental_compile()
```

Reply:
Yes, this is a known bug, cc @jackhumphries 

I think this check got added when Jack was looking into supporting multi-node. I think you could remove it, and it should be okay as long as all of the remote readers are on the same node.

Reply:
> all of the remote readers are on the same node.

Why should all remote readers be on the same node?

Reply:
Right now that is required because all of the remote readers will share one channel. But with your PR, we can create multiple channels, one for each unique reader node.

Reply:
Got it. I will update my PR. Currently, it only creates one `Channel` for all remote readers (readers which are not on the same actor as the writer).

Reply:
Yes, this is currently something I need to work on. The hope is that this PR will make it easy at this point to add support for readers across multiple nodes, though you don't need to explicitly support that in this PR--I will take care of it once this PR is merged.

Reply:
Chat with @jackhumphries offline. He will work on this part. I will not include the multi-node support in this PR.